plugins/inventory/inventory.py

```diff
@@ -0,0 +1,162 @@
+    version_added: 1.0.0
+  state_file:
+    description:
+      - Absolute path to an existing Terraform state file whose outputs will be listed.
+      - If I(state_file) and I(project_path) are not specified, C(terraform.tfstate) file in the current
+        working directory will be used.
+    type: path
+    version_added: 1.0.0
+  binary_path:
+    description:
+      - The path of a terraform binary to use.
+    type: path
+    version_added: 1.0.0
+"""
+
+EXAMPLES = r"""
+# Example that creates an inventory from terraform.tfstate file in cwd
+
+plugin: cloud.terraform.inventory
+
+
+# Example that creates an inventory from mycustomstate.tfstate file in selected project directory
+
+plugin: cloud.terraform.inventory
+project_path: some/project/path
+state_file: mycustomstate.tfstate
+"""
+
+
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+import yaml
+import json
+import subprocess
+from ansible.errors import AnsibleError
+from ansible.module_utils.common import process
+import os
+
+
+# TODO: remove when ansible provider is available
+TESTING_STATE_FILE = os.environ.get('TESTING_STATE_FILE', False)
+
+
+def get_state_content(terraform_binary, project_path, state_file):
+    command = [
+        terraform_binary,
+        "show",
+        "-json",
+        state_file if state_file else "terraform.tfstate"
+    ]
+
+    completed_process = subprocess.run(command, cwd=project_path, capture_output=True, check=False)
+
+    if completed_process.returncode != 0:
+        raise AnsibleError(
+            "Failure when getting Terraform state file. Exited {0}.\nstdout: {1}\nstderr: {2}".format(
+                completed_process.returncode, completed_process.stdout, completed_process.stderr
+            )
+        )
+
+    return json.loads(completed_process.stdout)
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+    NAME = "inventory"  # used internally by Ansible, it should match the file name but not required
+
+    def read_config_data(self, path):
+        """
+        Reads and validates the inventory source file and environ,
+        storing the provided configuration as options.
+        """
+        with open(path, "r") as inventory_src:
+            cfg = yaml.safe_load(inventory_src)
+        return cfg
+
+    def verify_file(self, path):
+        """
+        return true/false if this is possibly a valid file for this plugin to consume
+        """
+        with open(path, "r") as inventory_src:
+            config_contents = yaml.safe_load(inventory_src)
+        plugin = config_contents.get("plugin")
+        if not plugin:
+            return False
+        if plugin not in [self.NAME, "cloud.terraform.inventory"]:
+            return False
+        return True
+
+    def parse(self, inventory, loader, path, cache=False):
+        super(InventoryModule, self).parse(inventory, loader, path)
+
+        cfg = self.read_config_data(path)  # os.environ is not needed
```

Comment:
I don't understand what the comment means here. Of course you don't need to read the environment, it's a config file :)

Reply:
It was just that all config is read from the file, nothing is needed from env. The comment was for me, just forgot to delete it